src/terraform/mlz/variables.tf

```diff
@@ -59,6 +59,11 @@ variable "create_assignment" {
   default     = false
 }
 
+variable "deploymenttype" {
+  description = "Metadata corresponding with the deployment method."
+  type        = string
+  default     = "MissionLandingZoneTF"
+}
```

Comment:
Instead of adding another variable, what if we instead replace `deploymenttype` and `deploymentname` with a [map typed variable](https://www.terraform.io/docs/language/expressions/type-constraints.html#map-) called `tags` that we can default to the values we want? This also allows for the user to override the `tags` value with whatever map they want to specify at deployment time. Here's what that could look like:

```terraform
# somewhere in variables.tf
variable "tags" {
  description = "A map of key value pairs to apply as tags to resources provisioned in this deployment"
  type        = map(string)
  default = {
    "DeploymentName" : "mlz-dev"
    "DeploymentType" : "MissionLandingZoneTF"
  }
}
```

Then, in our Terraform modules, rather than specifying this map object to the value of `tags` we can just pass `var.tags`:

```terraform
# somewhere in main.tf

# instead of this implementation that specifies a map(string)
tags = {
  DeploymentName = var.deploymentname,
  DeploymentType = var.deploymenttype
}

# we'd use this
tags = var.tags
```

Then, at deployment time, the user could override this value (or any others they'd like to customize) by [overriding variables inline at the command-line](https://www.terraform.io/docs/language/values/variables.html#variables-on-the-command-line), or by [overriding variables with a .tfvars definition](https://www.terraform.io/docs/language/values/variables.html#variable-definitions-tfvars-files).

Reply:
I had tried that approach first, but there is yet another `deploymentname` parameter that would be in conflict with this in the variables.tf file that contains: 

```
variable "deploymentname" {
  description = "A name for the deployment. It defaults to dev."
  type        = string
  default     = "dev"
}
```

Unable to circularly reference this variable in the way that works with tagging in terraform. 
This might break the existing tag that is set in place?

Reply:
Let's replace `deploymentname` with this new variable too. It appears it's only used in assignment for the `tags` value on resources.

Reply:
resolving this conversation as proposed changes have been implemented and tested, new commit bc30649 is pending approval.